spreadsheet/formula/expression.go

```diff
@@ -10,4 +10,6 @@ package formula
 type Expression interface {
```

Comment:
Can you add a comment what an Expression is?  It seems to make sense that an Expression can be evaluated and has a reference.  I am not sure if it makes sense that it can be moved left?
Would it make sense to have another interface such as Movable that has MoveLeft ?  It would make sense that cells can be moved left, columns, not sure about other things?

Reply:
We apply MoveLeft for results of formula.ParseString(formStr string) which returns the Expression (and formula.ParseString is used also in other places in codebase where it should return Expression), so maybe just rename MoveLeft to anything else? Say, UpdateReferencesAfterRemoveColumn, or something shorter

Reply:
MoveLeft is ok i think, if it actually moves things left.  Need to document the MoveQuery also and probably define it with the interface?
It would be nice if we could avoid exporting those functions, MoveLeft sounds like a pretty funny thing to need to implement for an Expression, if one was creating their own Expression in an external package.

Please also add a comment on what an Expression is above the type, and examples of what it can be.  It seems like it can represent a cell, group of cells, etc...